### ISSUE TITLE
onClosed is not being called

### ISSUE THREAD

**Issue by fernandohur** [wunderlist/android-sliding-layer-lib#28]
I'm not doing nothing fancy, but OnInteractListener.onClosed() is not being called, however onClose() is. ¿Has anyone had this problem?

**Comment by AAverin** [comment]
Was this issue resolved? Looks like it's been about 2 years already.

**Comment by JlUgia** [comment]
I was not able to ever reproduce this. Can @fernandohur help us with that? Are you facing it @AAverin ?

**Comment by AAverin** [comment]
Yes, in some cases onClosed isn't called even though onClose is being invoked during animation. 
I think issue only reproduces on older devices like 4.2, and is related to animation in general not  being fully stable on the platform. 
Another case might be threading - I noticed there is no synchronization on methods that notify listeners

**Comment by JlUgia** [comment]
Can you reproduce faithfully? Would love to test myself. The synchronization on notifiers is a good point. Do you want to propose a PR?

**Comment by AAverin** [comment]
I'll see if I will have time to work on that, for now I'm a bit in a hurry with implementation.

**Comment by JlUgia** [comment]
I'll see if I can reproduce it in the meantime. Any help is more than welcome.
